### ISSUE TITLE
OF no longer working without JavaScript

### ISSUE THREAD

**Issue by soxet** [buckket/ofgrab#8]
Hi, I'm trying to run ofgrab for the first time but I'm getting
`[!] No posts found, there seems to be a problem`

I've tried various sessions, browsers and OSes

The command I'm running
`./ofgrab.py abcdefg qwertyiopasdf "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.14; rv:67.0) Gecko/20100101 Firefox/67.0"`

**Comment by UltimaHoarder** [comment]
"We're sorry but OnlyFans doesn't work properly without JavaScript enabled"

This is the error I get when I make a get request to the site with Python

**Comment by buckket** [comment]
Yeah OF seems to have jumped the shark. They are now using JS to assemble the whole HTML content in the client’s browser by side-loading it via encoded JS hunks.

The only options for scraping their content now is to either:
- use headless versions of Firefox/Chrome and execute their JS code
- implement the scraper as a browser extension
- try to extract the media URLs from the JS blob without executing it

None of these seem very appealing or straight forward. That’s why I will stop supporting this script and archive the repository until further notice. Sorry.

**Comment by UltimaHoarder** [comment]
I have a working script you can use as a reference @buckket 
https://github.com/DIGITALCRIMINAL/OnlyFans

OnlyFans now uses an API to get post data
It looks like this

> "https://onlyfans.com/api2/v2/users/"+user_id+"/posts?limit=10&offset=0&order=publish_date_desc&app-token="+app_token+""

user_id is the numerical id of the profile you want to scrape
app-token is the app-token .-.

To get it you can go to an OnlyFans account (yours even) and open your devtool dashboard and check the network tab to see your requests

**Comment by buckket** [comment]
@DIGITALCRIMINAL Thanks for the hint! Missed that one. Then it’s only a matter of finding out how to use their new API go get all the stuff we want. Will look into it!

**Comment by mizuoppai** [comment]
Are there any updates on this issue? Is there anything a non-coder can do to help?
